src/quark-core/src/QuarkScript.sol

```diff
@@ -118,4 +120,54 @@ abstract contract QuarkScript {
             sstore(key, value)
         }
     }
+
+    // Note: this may not be accurate after any nested calls from a script
+    function getActiveNonce() internal returns (bytes32) {
+        QuarkWallet self = QuarkWallet(payable(address(this)));
+
+        bytes32 activeNonceSlot = self.ACTIVE_NONCE_SLOT();
+        bytes32 value;
+        assembly {
+            value := sload(activeNonceSlot)
+        }
+        if (value == bytes32(0)) {
+            revert InvalidActiveNonce();
+        }
+
+        return value;
+    }
+
+    // Note: this may not be accurate after any nested calls from a script
+    function getActiveSubmissionToken() internal returns (bytes32) {
+        QuarkWallet self = QuarkWallet(payable(address(this)));
+
+        bytes32 activeSubmissionTokenSlot = self.ACTIVE_SUBMISSION_TOKEN_SLOT();
+        bytes32 value;
+        assembly {
+            value := sload(activeSubmissionTokenSlot)
+        }
+        if (value == bytes32(0)) {
+            revert InvalidActiveSubmissionToken();
+        }
+        return value;
+    }
+
+    // Note: this may not be accurate after any nested calls from a script
+    // Returns the active replay count of this script. Thus, the first submission should return 0,
+    // the second submission 1, and so on. This must be called before the script makes any external calls.
+    function getActiveReplayCount() internal returns (uint256) {
+        bytes32 nonce = getActiveNonce();
+        bytes32 submissionToken = getActiveSubmissionToken();
+        uint256 n;
+
+        if (submissionToken == bytes32(type(uint256).max)) {
```

Comment:
can we not do `QuarkNonceManager.EXHAUSTED`?

the tiny workaround here is you can do what we did in `QuarkWallet` with the `QuarkWalletMetadata` library:

```solidity
library SpecialNonce {
  bytes32 internal constant FREE = bytes32(0);
  bytes32 internal constant EXHAUSTED = bytes32(type(uint256).max);
}
```

and then `SpecialNonce.EXHAUSTED` gets inlined everywhere. 🤷